more_itertools/more.py

```diff
@@ -2456,19 +2455,31 @@ def __getitem__(self, key):
         raise TypeError('islice_extended.__getitem__ argument must be a slice')
 
 
-def _islice_helper(it, s):
-    start = s.start
-    stop = s.stop
-    if s.step == 0:
+def _islice_helper(iterable, s):
+    start, stop, step = s.start, s.stop, s.step
```

Comment:
Same question here - why switch to tuple unpacking here? It's not really related to the change being proposed.